src/lib.rs

```diff
@@ -187,6 +188,11 @@ pub mod pallet {
 		#[pallet::constant]
 		type StakeUnlockDelay: Get<BlockNumberFor<Self>>;
 
+		/// Number of blocks to wait before reusing funds previously assigned to a collator.
+		/// It should be set to at least one session.
```

Comment:
Hmmmmm then you could restake during the same session. This brings the following "attack" vector:

A user removes all the stake upon a new session starts. It does not care if the delay is less than one session. Once the session is about to end so that he can identify the most performant collators, and places the stake on them.

This is why the minimum period should be at least until **there is a new session**. This way the same case applies, but at that time the user has already lost the rewards of a whole session, which is a huge penalty for such a little extra reward. 

However, session lengths are not necessarily deterministic, and that's why I suggest at least a whole one.